### PR TITLE
Added a fallback to both the player and workstation drivers from the vmware builder when trying to determine the network-mapping configuration

### PR DESCRIPTION
**Pull request by arizvisa**
When installing VMWare Workstation on Linux hosts, in some instances the installer will not generate the network-mapping configuration (netmap.conf) at the time of installation. As a workaround, users have suggested to explicitly run the network configuration and commit their changes in order for the network-mapping configuration to be written.

On my instance of VMWare Workstation (16.0.0 build-16894299) the installer actually creates a "networking" configuration file (which uses the syntax that the builders are parsing for for VMWare Fusion), and then after configuring the network the "netmap.conf" file is created. As this "networking" file contains similar information, it can also be used to infer the network-mapping configuration as a fallback in the situation where the "netmap.conf" file hasn't been created yet.

To take advantage of this, both the player and workstation drivers for the vmware builder were modified to determine which file is available. The original logic is still given priority in that if the original "netmap.conf" file is found, its parser is used to return type which implements the `NetworkNameMapper` interface. In this original logic, another case was added so that if the "netmap.conf" was not found, then the "networking" file is checked if it exists. If this file is found, then the parser for the "networking" file is used. If neither was found, an error is returned with a complaint and the paths that it checked for.

I'm not sure whether the symptoms reported in #10009 are the same as on Windows (as nobody had confirmed it), but I kind of figured that it'd be better to be defensive about it and check with both methods anyways in case the network-mapping configuration can't be determined. Thus, this affects both the Windows and Linux platforms.

This fixes issue #10009.

### CODE REVIEW

**Comment by arizvisa** [comment]
Heh, I'm "attempting to deploy a commit to the HashiCorp Team on Vercel?" That sounds pretty aggressive...

Anyways, it seems that users are okay with working around this issue manually. So it's moreso submitted for y'all to review..

**Comment by azr** [comment]
Hey @arizvisa about Vercel nothing to worry about; the docs deployer detected that a file deemed as highly sensitive was deleted and it won’t create a deploy preview w/o approval if an outside contributor tries to edit the file. Even though no docs were edited.

I think this is probably due to the fact that you have branched from an old-ish commit.

File: builder/vmware/common/driver_player5.go [hunk 1]
```diff
@@ -205,11 +205,28 @@ func (d *Player5Driver) Verify() error {
 	d.VmwareDriver.NetworkMapper = func() (NetworkNameMapper, error) {
 		pathNetmap := playerNetmapConfPath()
 		if _, err := os.Stat(pathNetmap); err != nil {
```
Comment (SwampDragons) [inline builder/vmware/common/driver_player5.go:207]: Don't we need to change this conditional to the == nil?

**Review by SwampDragons — CHANGES_REQUESTED** [review]
One tiny typo, otherwise this looks good Thanks!

File: builder/vmware/common/driver_player5.go (hunk 1 above)
Comment (arizvisa) [reply builder/vmware/common/driver_player5.go:207]: good catch.

**Comment by arizvisa** [comment]
> One tiny typo, otherwise this looks good Thanks!

Fixed that type-o and documented the logic to highlight the explicit checking for a non-error.

**Comment by arizvisa** [comment]
Btw, I wasn't sure if the file that I'm opening actually exists on windows..but it does on linux. I asked in the original issue if there was a difference, but nobody responded. Although we're still using the original methodology, so the original workaround that was proposed (to run the network config and close it) will still be valid in getting vmware to create the "netmap.conf" file.

Anyways, the drivers implement functions like "workstationNetmapConfPath", "playerNetmapConfPath" to determine the correct filename for this stuff. However since this was a fallback-type of situation, I instead hardcoded the path to the "networking" file. I haven't confirmed whether this file exists on Windows, so there's a chance that this PR is ineffective for that platform.

So, if users still complain in the future...then that path to the "networking" file will need to be corrected to wherever it's supposed to be located on Windows, and probably a new function will need to be added to the drivers to "find" the correct path to the file.

**Comment by SwampDragons** [comment]
That's fine. Since the read is protected by a file stat, it's not going to do any harm as coded. Thanks for this!

**Pull request merged**